src/combine-results.ts

```diff
@@ -1,92 +1,95 @@
 import fs from "fs";
 import path from "path";
 
-import * as artifact from "@actions/artifact";
-import * as core from "@actions/core";
-import * as exec from "@actions/exec";
-import * as github from "@actions/github";
-import * as io from "@actions/io";
+import { create as createArtifactClient } from "@actions/artifact";
+import { getInput, warning, setFailed } from "@actions/core";
+import { getExecOutput } from "@actions/exec";
+import { context, getOctokit } from "@actions/github";
+import { mkdirP, mv } from "@actions/io";
+
+const formatBody = (query: string, results: string): string => `# Query
+<details>
+  <summary>Click to expand</summary>
+
+\`\`\`ql
+${query}
+\`\`\`
+</details>
+
+# Results
+
+|Repository|Results|
+|---|---|
+${results}`;
 
 async function run(): Promise<void> {
   try {
-    const query = core.getInput("query", { required: true });
-    const language = core.getInput("language", { required: true });
-    const token = core.getInput("token", { required: true });
+    const query = getInput("query", { required: true });
+    const language = getInput("language", { required: true });
+    const token = getInput("token", { required: true });
 
-    await io.mkdirP("artifacts");
-    const artifactClient = artifact.create();
+    await mkdirP("artifacts");
+    const artifactClient = createArtifactClient();
     const downloadResponse = await artifactClient.downloadAllArtifacts(
       "artifacts"
     );
 
-    await io.mkdirP("results");
+    await mkdirP("results");
 
-    const context = github.context;
-    const octokit = github.getOctokit(token);
+    const octokit = getOctokit(token);
     const title = `Query run by ${context.actor} against ${downloadResponse.length} \`${language}\` repositories`;
     const issue = await octokit.rest.issues.create({
       owner: context.repo.owner,
       repo: context.repo.repo,
       title,
     });
 
-    let body = `# Query
-<details>
-  <summary>Click to expand</summary>
-
-\`\`\`ql
-${query}
-\`\`\`
-</details>
-
-# Results
+    const csvs: string[] = [];
+    const results = await Promise.all(
+      downloadResponse.map(async function (response) {
```

Comment:
Given there is a lot of async stuff here that doesn't depend on each other, simply collecting the results in order as resolved promises, doing an async map makes a lot of sense.

If the order of `csvs` also matters, then we'd need to perhaps return an object and collect those values in order as well - and then iterate the response - but assuming that this function has a lot of stuff that will take a while async, it seems like a good opportunity to do it async.

I make the assumption that `csvs` doesn't need to be ordered, and that the issue comments don't need to be guaranteed to be in order either, but obviously those assumptions needs to be confirmed before this could be accepted.

Reply:
Yes, it should be safe.